src/base/http/requestparser.cpp

```diff
@@ -30,314 +31,307 @@
 #include "requestparser.h"
 
 #include <QDebug>
-#include <QDir>
+#include <QRegularExpression>
 #include <QStringList>
 #include <QUrl>
 #include <QUrlQuery>
 
-const QByteArray EOL("\r\n");
-const QByteArray EOH("\r\n\r\n");
+#include "base/utils/string.h"
 
-inline QString unquoted(const QString &str)
+using namespace Http;
+using QStringPair = QPair<QString, QString>;
+
+namespace
 {
-    if ((str[0] == '\"') && (str[str.length() - 1] == '\"'))
-        return str.mid(1, str.length() - 2);
+    const QByteArray EOH = QByteArray(CRLF).repeated(2);
 
-    return str;
-}
+    QByteArray chopIfEndsWith(QByteArray in, const QByteArray &str)
+    {
+        if (in.endsWith(str))
+            in.chop(str.size());
+        return in;
+    }
 
-using namespace Http;
+    QList<QByteArray> split(const QByteArray &in, const QByteArray &sep, const QString::SplitBehavior behavior = QString::KeepEmptyParts)
+    {
+        // mimic QString::split(sep, behavior)
 
-RequestParser::ErrorCode RequestParser::parse(const QByteArray &data, Request &request, uint maxContentLength)
-{
-    return RequestParser(maxContentLength).parseHttpRequest(data, request);
+        if (sep.isEmpty())
+            return {};
+
+        QList<QByteArray> ret;
+
+        int head = 0;
+        while(head < in.size()) {
+            int end = in.indexOf(sep, head);
+            if (end < 0)
+                end = in.size();
+
+            // omit empty parts
+            const QByteArray part((in.constData() + head), (end - head));
+            if (!part.isEmpty() || (behavior == QString::KeepEmptyParts))
+                ret += part;
+
+            head = end + sep.size();
+        }
+
+        return ret;
+    }
 }
 
-RequestParser::RequestParser(uint maxContentLength)
-    : m_maxContentLength(maxContentLength)
+RequestParser::RequestParser()
+    : m_request()
 {
```

Comment:
you mean the `m_request` initialization?

Reply:
Yes.

Reply:
isn't explicitly initializing all class members a good practice?

Reply:
It doesn't make any sense to me. Same thing to do like `Class obj = Class();` instead of `Class obj;`.
Initializing makes sense in the following cases:
1. Simple types (like bool, int, double etc.) which may remain uninitialized in other case.
2. Non-default initialization of class types.
3. Initialization based on some parameters.

Reply:
will remove.